pkgs/by-name/sc/scalpel/package.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, fetchFromGitHub
+, stdenv
+, autoconf
+, automake
+, libtool
+, tre
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "scalpel";
+  version = "2.1";
+
+  src = fetchFromGitHub {
+    owner = "sleuthkit";
+    repo = "scalpel";
+    rev = "35e1367ef2232c0f4883c92ec2839273c821dd39";
+    hash = "sha256-0lqU1CAcWXNw9WFa29BXla1mvABlzWV+hcozZyfR0oE=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    libtool
+    tre
+  ];
+
+  postPatch = ''
+    sed -i \
+      -e 's|#define\s*SCALPEL_DEFAULT_CONFIG_FILE\s.*"scalpel.conf"|#define SCALPEL_DEFAULT_CONFIG_FILE "${placeholder "out"}/share/scalpel/scalpel.conf"|' \
+      src/scalpel.h
+  '';
+
+  env.CXXFLAGS = "-std=c++14 -Wno-error=reserved-user-defined-literal";
+
+  preConfigure = ''
+    ./bootstrap
+  '';
+
+  configureFlags = [
+    "--with-pic"
+    "CXXFLAGS=-std=c++14"
+  ];
+
+  postInstall = ''
+    install -Dm644 scalpel.conf -t $out/share/scalpel/
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/sleuthkit/scalpel";
+    description = "Recover files based on their headers, footers and internal data structures, based on Foremost";
+    mainProgram = "scalpel";
+    maintainers = with maintainers; [ shard7 ];
+    platforms = platforms.unix;
+    sourceProvenance = with sourceTypes; [ fromSource binaryNativeCode ];
```

Comment:
```suggestion
```
when looking at result i do not see any binaries distributed -- looking at the source there are some dlls for windows but that's it. could you point out where the binaryNativeCode is?

Reply:
eg: everything that ends up in the nix store is compiled from source. which doesn't count as binaryNativeCode, I believe that `sourceProvenance` binaryNatvieCode is used when the just re-distributing compiled code and not building it. so I think that this line can be removed -- unless i am missing something.

Reply:
@a-n-n-a-l-e-e I thought [some of these files](https://github.com/sleuthkit/scalpel/tree/master/pthreads-win32) were needed as libs (especially [libpthreadGC2.a](https://github.com/sleuthkit/scalpel/blob/master/pthreads-win32/libpthreadGC2.a) and [libpthreadGCE2.a](https://github.com/sleuthkit/scalpel/blob/master/pthreads-win32/libpthreadGCE2.a). I'm going to check again.

Also, is it normal, that when I try building with the updated changes for the derivation, I get:
`cc1plus: error: '-Wno-error=reserved-user-defined-literal': no option '-Wreserved-user-defined-literal'` ?

Reply:
> @a-n-n-a-l-e-e I thought [some of these files](https://github.com/sleuthkit/scalpel/tree/master/pthreads-win32) were needed as libs (especially [libpthreadGC2.a](https://github.com/sleuthkit/scalpel/blob/master/pthreads-win32/libpthreadGC2.a) and [libpthreadGCE2.a](https://github.com/sleuthkit/scalpel/blob/master/pthreads-win32/libpthreadGCE2.a). I'm going to check again.

thanks. if that is the case it is loading the win32 blobs then i'd assume it is x86 only and aarch64 not supported.

> Also, is it normal, that when I try building with the updated changes for the derivation, I get: `cc1plus: error: '-Wno-error=reserved-user-defined-literal': no option '-Wreserved-user-defined-literal'` ?

argh. i guess that option is clang only. so need to add it conditionally.
```nix
env.CXXFLAGS = "-other flags" + lib.optionalString stdenv.cc.isClang "  -Wno-error=reserved-user-defined-literal";
```

Reply:
> @a-n-n-a-l-e-e I thought [some of these files](https://github.com/sleuthkit/scalpel/tree/master/pthreads-win32) were needed as libs (especially [libpthreadGC2.a](https://github.com/sleuthkit/scalpel/blob/master/pthreads-win32/libpthreadGC2.a) and [libpthreadGCE2.a](https://github.com/sleuthkit/scalpel/blob/master/pthreads-win32/libpthreadGCE2.a). I'm going to check again.

these do not seem to be needed, i just tested by adding.
```nix
 preBuild = ''
    rm -f pthreads-win32/*.dll pthreads-win32/*.a
  '';
```
and everything built fine on darwin

Reply:
@a-n-n-a-l-e-e I double checked, you are right, I'll remove `binaryNativeCode`. I re-built with the changes and it works. Thanks again.

Reply:
> @a-n-n-a-l-e-e I double checked, you are right, I'll remove `binaryNativeCode`. I re-built with the changes and it works. Thanks again.

remove the line please.
https://nixos.org/manual/nixpkgs/stable/#sec-meta-sourceProvenance
when not set built from source is implied.

Reply:
Why a check saying "package.nix" not existing is failing if it exists?

Reply:
looks like a recent commit https://github.com/NixOS/nixpkgs/pull/279892 might be causing this and there is a PR for the fix: https://github.com/NixOS/nixpkgs/pull/281366